crates/ra_parser/src/grammar/expressions/atom.rs

```diff
@@ -307,6 +313,16 @@ fn while_expr(p: &mut Parser, m: Option<Marker>) -> CompletedMarker {
 // fn foo() {
 //     for x in [] {};
 // }
+
+// test for_range_block
+// fn foo() {
+//    for x in 0 .. {
+//        break;
+//    }
+//    for _x in 0 .. (0 .. {1 + 2}).sum::<u32>() {
```

Comment:
... and this test to test data, as it doesn't test a single isolated bit of a grammar and is rather a "stress test".

Reply:
ok